src/app/wallets/get-pending-onchain-balance-for-wallet.ts

```diff
@@ -0,0 +1,52 @@
+import { BTC_NETWORK, ONCHAIN_MIN_CONFIRMATIONS, SECS_PER_10_MINS } from "@config"
+
+import { CacheKeys } from "@domain/cache"
+import { OnChainError, TxDecoder, TxFilter } from "@domain/bitcoin/onchain"
+
+import { baseLogger } from "@services/logger"
+import { RedisCacheService } from "@services/cache"
+import { OnChainService } from "@services/lnd/onchain-service"
+import { IncomingOnChainTxHandler } from "@domain/bitcoin/onchain/incoming-tx-handler"
+
+const redisCache = RedisCacheService()
+
+export const getPendingOnChainBalanceForWallet = async (
+  wallets: Wallet[],
+): Promise<{ [key: WalletId]: CurrencyBaseAmount } | ApplicationError> => {
+  const onChain = OnChainService(TxDecoder(BTC_NETWORK))
+  if (onChain instanceof OnChainError) {
+    baseLogger.warn({ onChain }, "impossible to create OnChainService")
+    return onChain
+  }
+
+  // we are getting both the transactions in the mempool and the transaction that
+  // have been mined by not yet credited because they haven't reached enough confirmations
+  const onChainTxs = await redisCache.getOrSet({
+    key: CacheKeys.LastOnChainTransactions,
+    ttlSecs: SECS_PER_10_MINS,
+    fn: () => onChain.listIncomingTransactions(ONCHAIN_MIN_CONFIRMATIONS),
```

Comment:
here, and in 2 other places we have `redisCache.getOrSet({
    key: CacheKeys.LastOnChainTransactions })`, we are pre-fetching in advance the onchain Service, and failing in case the onchain service is down.

I believe this feels like an anti pattern. There are 2 reasons we are using redis:
- we want to reduce the load on lnd, which we achieve by using redis here
- we want to be resilient of error momentarily in case of a lnd restart

because we fail early if onchain service is down, we're removing many of the upside of using a cache here.

probably the fetching of the onchain service should only initiated if the value is not in cache.

```suggestion
    fn: () => {
      const onChain = OnChainService(TxDecoder(BTC_NETWORK))
  if (onChain instanceof OnChainError) {
    baseLogger.warn({ onChain }, "impossible to create OnChainService")
    return onChain
  }
    onChain.listIncomingTransactions(ONCHAIN_MIN_CONFIRMATIONS),
    }
```

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1512/commits/eea915b7e389d52d9ee712c81b733a04f31e96e8 & https://github.com/GaloyMoney/galoy/pull/1512/commits/2c7b1344df4196582af7e748de047e67308dcda8